### ISSUE TITLE
Seeking advice for preprocess method

### ISSUE THREAD

**Issue by KrisZHH** [AlfredXiangWu/face_verification_experiment#139]
@AlfredXiangWu 
First, thanks for providing such light CNN model. I am trying to reproduce your project now.
I downloaded your LightenedCNN_C.caffemodel and used it extract feature on FaceScrub & Megaface dataset. But I found that the result was pretty low: the Rank-1 was only 61% while the TPR@10e-6 below 3% (both result under 10000 distractors).
I think must my preprocess have some problems. I use python API and my preprocess as below:
1. crop an aligned image into 128x128;
2. convert RGB image into grey;
3. transpose the 3rd dimension(channel dimension) to 1st.
The code like below:

`transformer = caffe.io.Transformer({'data': net.blobs['data'].data.shape})`
`transformer.set_transpose('data', (2,0,1)) `

`image = caffe.io.load_image(os.path.join(root, file)) `
`image = color.rgb2gray(image[8:136, 8:136]).reshape(128, 128, 1) `
`transformed_image = transformer.preprocess('data', image)`

`net.blobs['data'].data[...] = transformed_image `
`output = net.forward()`

`feat = net.blobs['eltwise_fc1'].data[0] `

I want to know whether there are some problems on my preprocess, or there are some preprocess step I missed.

Thank you. I am looking forward to your reply.

**Comment by AlfredXiangWu** [comment]
I think the problem is that we use different alignment method compared with the official aligned images.

I recommend you download the original images, and then you can do face detection and landmark detection for each face image. After obtaining the 5 facial landmarks, you can follow the alignment method in Readme. 

Besides, I strongly recommend you to train and test model by pytorch version. We also release the MegaFace features. 
https://github.com/AlfredXiangWu/LightCNN

**Comment by KrisZHH** [comment]
I didn't use the official aligned images. I used the 5 points landmarks method  either. 
I want to reconfirm whether my preprocess method is OK? And do I need to add any preprocess method?
If my image alignment method & preprocess without problem, which step do you think probably went wrong?
Thank you.

**Comment by yiakwy** [comment]
@AlfredXiangWu your explanation of preprocessing in Readme is nonsense, frankly. 

1. first convert to BGR image then grey level image. This is ok
2. "normlized to" correspond to the following two operations:
2.1  net.set_input_scale: (0,255) -> (0,1)
2.2 net.set_mean If one want to predicat face names from an image where figures from LFW dataset, what preprocessing should they do?  You should clearify why you used image mean from another dataset other than LFW.

I have reviewed your evaluation code, basically about comparing feature vectors. That explains why your model laster layer output is 'fc' not 'softmax'.

Hence, I think your incomplete codes are misleading people in a wrong way. For example, I don't care about face vector, I just to get labels of the people.
